### PR TITLE
Swope agent

### PR DESCRIPTION
**Pull request by swopebe**

### CODE REVIEW

File: clusters/cluster_lifecycle/ansible_job.adoc [hunk 1]
```diff
@@ -13,7 +13,7 @@ Your definition file contents might resemble the following example:
 +
 [source,yaml]
 ----
-apiVersion: apiVersion: tower.ansible.com/v1alpha1
+apiVersion: tower.ansible.com/v1alpha1
 kind: AnsibleJob
```
Comment (swopebe) [inline clusters/cluster_lifecycle/ansible_job.adoc:17]: Why we had two here, I don't know. Removed it but not part of this issue.

File: clusters/cluster_lifecycle/ansible_job.adoc [hunk 2]
```diff
@@ -46,7 +46,7 @@ clusterConditions:
 +
 You can either paste this into the YAML content of the placement rule or you can select the option to _Deploy to all online clusters and local cluster_ on the _Application create_ page of the console. 
 
-. Create or import your cluster by following the instructions in xref:../cluster_lifecycle/create_intro.adoc#create-intro[Cluster creation] or xref:../cluster_lifecycle/import_intro.adoc#import-intro[Cluster import], respectively.
+. Create or import your cluster by following the instructions in xref:../cluster_lifecycle/create_intro.adoc#create-intro[Creating clusters] or xref:../cluster_lifecycle/import_intro.adoc#import-intro[Cluster import], respectively.
```
Comment (swopebe) [inline clusters/cluster_lifecycle/ansible_job.adoc:49]: Not sure why we named this "Cluster creation" and not an active "Creating clusters" (legacy content not written by the team as it is; changing this.

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[#config-agent-proxy]
+= Configuring _AgentClusterInstall_proxy
+
+Proxy defines the proxy settings used for the install configuration
```
Comment (swopebe) [inline clusters/cluster_lifecycle/cim_config_agent_proxy.adoc]: I need a better description, prereqs

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc [hunk 4]
```diff
@@ -0,0 +1,31 @@
+[#config-agent-proxy]
+= Configuring _AgentClusterInstall_ proxy
+
+The proxy defines the proxy settings used for the configuration.
+
```
Comment (swopebe) [inline clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:5]: We need a more informative description

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+proxy:
+  httpProxy: http://<username>:<password>@<proxy.example.com>:<port> <1>
+  httpsProxy: https://<username>:<password>@<proxy.example.com>:<port> <2>
+  noProxy: <wildcard-of-domain>,<provisioning-network/CIDR>,<BMC-address-range/CIDR> <3>
+----
+<1> `httpProxy` is the URL of the proxy for HTTP requests. Replace the username and password values with your credentials for your proxy server. Replace `proxy.example.com` with the path of your proxy server.
+<2> `httpsProxy` is the URL of the proxy for HTTPS requests. Replace the values with your credentials. Replace `port` with the communication port with the proxy server.
+<3> `noProxy`	is a comma-separated list of domains and CIDRs for which the proxy should not be used. Replace `wildcard-of-domain` with an entry for domains that should bypass the proxy. Replace `provisioning-network/CIDR` with the IP address of the provisioning network and the number of assigned IP addresses, in CIDR notation. Replace `BMC-address-range/CIDR` with the BMC address and the number of addresses, in CIDR notation.
+
+. Additional step here?
```
Comment (swopebe) [inline clusters/cluster_lifecycle/cim_config_agent_proxy.adoc]: Please help with the rest of the procedure. (Apply the file?)

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+[#resources-creating-cluster-proxy]
+== Additional resources
```
Comment (swopebe) [inline clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:26]: Where else do we need to send the user?

File: clusters/main.adoc [hunk 7]
```diff
@@ -52,6 +52,7 @@ include::cluster_lifecycle/create_vm.adoc[leveloffset=+4]
 include::cluster_lifecycle/create_openstack.adoc[leveloffset=+4]
 include::cluster_lifecycle/create_cluster_on_prem.adoc[leveloffset=+4]
 include::cluster_lifecycle/create_proxy_env.adoc[leveloffset=+4]
+include::cluster_lifecycle/cim_config_agent_proxy.adoc[leveloffset=+4]
```
Comment (swopebe) [inline clusters/main.adoc:55]: file added to main by create proxy.

File: clusters/cluster_lifecycle/create_intro.adoc [hunk 8]
```diff
@@ -13,4 +13,5 @@ Learn how to create {ocp} clusters across cloud providers with {mce-short}.
 * xref:../cluster_lifecycle/create_vm.adoc#creating-a-cluster-on-vmware-vsphere[Creating a cluster on VMware vSphere]
 * xref:../cluster_lifecycle/create_openstack.adoc#creating-a-cluster-on-openstack[Creating a cluster on Red Hat OpenStack Platform]
 * xref:../cluster_lifecycle/create_cluster_on_prem.adoc#creating-a-cluster-on-premises[Creating a cluster in an on-premises environment]
-* xref:../hosted_control_planes/hosted_intro.adoc#hosted-control-planes-intro[Hosted control planes]
+* xref:../cluster_lifecycle/create_proxy_env.adoc#creating-a-cluster-proxy[Creating a cluster in a proxy environment]
+* xref:../cluster_lifecycle/cim_config_agent_proxy_.adoc#config-agent-proxy[Configuring _AgentClusterInstall_ proxy]
```
Comment (swopebe) [inline clusters/cluster_lifecycle/create_intro.adoc:17]: Added to create intro, I think that is what we said?

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 4 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:5]: The agent cluster install proxy fields determine the proxy settings at install time and are used to create the cluster-wide proxy resource in the resulting cluster.

If, for example, someone only needed a proxy for workloads on the cluster, but not before install they could set just the cluster proxy and not the one on the infraenv (but this is probably too much info to put in the doc).

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 5 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc]: This isn't a full/valid `AgentClusterInstall` so I'd hesitate to put a step explicitly saying to `oc apply` this.

File: clusters/cluster_lifecycle/ansible_job.adoc (hunk 2 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/ansible_job.adoc:49]: Maybe https://github.com/stolostron/rhacm-docs/blob/2.12_stage/clusters/cluster_lifecycle/cim_create_cli.adoc#creating-a-host-inventory which explains the API for proxy on an infraenv (which is specifically for before install begins and will not add the proxy fields to the resulting cluster)

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 6 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:26]: Maybe https://github.com/stolostron/rhacm-docs/blob/2.12_stage/clusters/cluster_lifecycle/cim_create_cli.adoc#creating-a-host-inventory which explains the API for proxy on an infraenv (which is specifically for before install begins and will not add the proxy fields to the resulting cluster)

File: clusters/cluster_lifecycle/create_intro.adoc (hunk 8 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/create_intro.adoc:17]: Yeah, this seems fine to me

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 4 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:5]: >If you only need a proxy for workloads on the cluster, but not before installation, you can configure cluster proxy.

You'd need the cluster proxy here in both cases.  Configuring it in the infraenv doesn't handle both situations.

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
++
+[source,yaml]
+----
+apiVersion: extensions.hive.openshift.io/v1beta1
+kind: AgentClusterInstall
+proxy:
+  httpProxy: http://<username>:<password>@<proxy.example.com>:<port> <1>
+  httpsProxy: https://<username>:<password>@<proxy.example.com>:<port> <2>
+  noProxy: <wildcard-of-domain>,<provisioning-network/CIDR>,<BMC-address-range/CIDR> <3>
```
Comment (carbonin) [inline clusters/cluster_lifecycle/cim_config_agent_proxy.adoc]:
```suggestion
apiVersion: extensions.hive.openshift.io/v1beta1
kind: AgentClusterInstall
spec:
  proxy:
    httpProxy: http://<username>:<password>@<proxy.example.com>:<port> <1>
    httpsProxy: https://<username>:<password>@<proxy.example.com>:<port> <2>
    noProxy: <wildcard-of-domain>,<provisioning-network/CIDR>,<BMC-address-range/CIDR> <3>
```

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 4 above)
Comment (swopebe) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:5]: @carbonin ok, can you provide a better description or should we just remove it? Assume I know nothing about cluster proxy and agent cluster install. We simply have to keep up with too many components. So I will need more information about this for the description.

**Review by oafischer — APPROVED** [review]
Looks good from a doc perspective.

File: clusters/cluster_lifecycle/cim_config_agent_proxy.adoc (hunk 4 above)
Comment (carbonin) [reply clusters/cluster_lifecycle/cim_config_agent_proxy.adoc:5]: I think we can just remove the second sentence entirely.

**Pull request merged**